quiche/src/lib.rs

```diff
@@ -4670,10 +4680,26 @@ impl Connection {
             return Ok(());
         }
 
+        let is_writable = stream.is_writable();
+
         stream.urgency = urgency;
         stream.incremental = incremental;
 
-        // TODO: reprioritization
+        let old_priority_key = stream.priority_key.clone();
```

Comment:
I wonder if something like the following would work:
```rust
let new_priority_key = Arc::new(...);
let old_priority_key = std::mem::replace(&mut stream.priority_key, new_priority_key);

self.streams.writable_prioritized_update(
  &old_priority_key,
  &stream.priority_key,
  is_writable,
);
```
Looks a little cleaner (if it works).

Reply:
yeah this works